knowledge-base/how-to-convert-svg-to-png.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Covert SVG graphics to PNG picture
+description: This article demonstrates how to display SVG graphics in PictureBox by converting it to PNG
+type: how-to
+page_title: How to convert SVG to PNG 
+slug: how-to-convert-svg-to-png
+position: 
+tags: 
+ticketid: 1400737
+res_type: kb
+---
+
+## Environment
+<table>
+	<tr>
+		<td>Product Version</td>
+		<td>13.0.19.222</td>
+	</tr>
+	<tr>
+		<td>Product</td>
+		<td>Progress® Telerik® Reporting</td>
+	</tr>
+</table>
+
+
+## Description
+Currently, the [PictureBox](../report-items-picture-box) can hold only the formats supported by GDI+ (BMP, GIF, JPEG, EXIF, PNG, and TIFF). This KB article demonstrates how to convert SVG graphics to PNG picture through the examples below. Note that a  third party package- Svg should be added to the with NuGet packages.  
+
+## Solution
+The first option is by using [UserFunction](../expressions-user-functions) when the report is developed from the Standalone designer. It can be implemented as the code snippet below:
+```
+public static Bitmap ConvertToBitmap(string url)
+{
+    var svgDocument = Svg.SvgDocument.Open(url);
+    svgDocument.ShapeRendering = SvgShapeRendering.Auto;
+ 
+    Bitmap bmp = svgDocument.Draw(120, 120); // Draw Bitmap in any Size you need - for example 120px x 120px
+    return bmp;
+}
+```
+Copy the generated assembly (i.e. UserFunction.dll), as well as the one for Svg to the folder that contains the Report Designer - (Telerik Reporting installation folder)\Report Designer (e.g. C:\Program Files (x86)\Progress\Telerik Reporting R1 2019\Report Designer).
+Next, it will be necessary to register the copied assemblies in the Telerik.ReportDesigner.exe.config file. For the purpose you should add the following piece of code in the configuration file: 
+```
+<Telerik.Reporting>
+   <AssemblyReferences>
+     <add name="UserFunction" version="1.0.0.0" culture="neutral" publicKeyToken="null" />
+     <add name="Svg" version="2.4.2.26702" culture="neutral" publicKeyToken="null" />
+   </AssemblyReferences>
+ </Telerik.Reporting>
+```
+Then the value of the picture box should be set to the following expression: 
+
+```= UserFunction.Custom.ConvertToBitmap(Parameters.url.Value)```, where the report parameter holds the picture destination.
+
+The second example is when the report is designed in Visual Studio. The following code snippet should be added to ItemDataBound or ItemDataBinding event of the PictureBox:
```

Comment:
In general, we do not recommend using report events. For visual studio report designer should be the same approach (with user function). With the advantage that the user function could be a part of the current report class - check https://docs.telerik.com/reporting/expressions-user-functions